envoy-control-core/src/main/resources/lua/egress_auto_service_tags.lua

```diff
@@ -0,0 +1,25 @@
+function envoy_on_request(handle)
+    rejectServiceTagDuplicatingAutoServiceTag(handle)
+end
+
+function rejectServiceTagDuplicatingAutoServiceTag(handle)
+    local autoServiceTagPreference = handle:metadata():get("auto_service_tag_preference")
+    if autoServiceTagPreference == nil then
+        return
+    end
+    local requestServiceTag = (handle:streamInfo():dynamicMetadata():get("envoy.lb") or {})["%SERVICE_TAG_METADATA_KEY%"]
+    if (requestServiceTag == nil) then
+        return
+    end
+
+    for i = 1, #autoServiceTagPreference do
+        if requestServiceTag == autoServiceTagPreference[i] then
+            local message = "Request service-tag '" .. requestServiceTag .. "' duplicates auto service-tag preference. "
+                .. "Remove service-tag parameter from the request"
```

Comment:
maybe it should be removed in lua filter, and only logged for service owners, instead of 400.

Reply:
Discussed on Zoom. Will keep it as it is, because this is for independent feature `autoServiceTag` which we want to remove in the future, we don't want to change it.